### PR TITLE
Make retribert tokenizers independent from BertTokenizer

### PR DESCRIPTION
**Pull request by Davidy22**
Part of a series of commits to step towards resolving #19303

# What does this PR do?

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Davidy22** [comment]
Are these line length check errors coming from the comments? Those are the only lines that approach the cusp of 119, and my editor tells me they're right under the line

File: src/transformers/models/retribert/tokenization_retribert.py [hunk 1]
```diff
@@ -40,18 +45,479 @@
...
     r"""
     Constructs a RetriBERT tokenizer.
 
     [`RetriBertTokenizer`] is identical to [`BertTokenizer`] and runs end-to-end tokenization: punctuation splitting
     and wordpiece.
 
-    Refer to superclass [`BertTokenizer`] for usage examples and documentation concerning parameters.
+    Args:
+    This tokenizer inherits from [`PreTrainedTokenizer`] which contains most of the main methods. Users should refer to:
+    this superclass for more information regarding those methods.
```
Comment (sgugger) [inline src/transformers/models/retribert/tokenization_retribert.py:80]: The args keyword is not properly placed, it should be below the intro:
```suggestion
    This tokenizer inherits from [`PreTrainedTokenizer`] which contains most of the main methods. Users should refer to:
    this superclass for more information regarding those methods.
    
    Args:
```

File: src/transformers/models/retribert/tokenization_retribert_fast.py [hunk 2]
```diff
@@ -46,14 +51,46 @@
...
     r"""
     Construct a "fast" RetriBERT tokenizer (backed by HuggingFace's *tokenizers* library).
 
     [`RetriBertTokenizerFast`] is identical to [`BertTokenizerFast`] and runs end-to-end tokenization: punctuation
     splitting and wordpiece.
 
-    Refer to superclass [`BertTokenizerFast`] for usage examples and documentation concerning parameters.
+    Args:
+    This tokenizer inherits from [`PreTrainedTokenizerFast`] which contains most of the main methods. Users should
+    refer to this superclass for more information regarding those methods.
```
Comment (sgugger) [inline src/transformers/models/retribert/tokenization_retribert_fast.py:63]:
```suggestion
    This tokenizer inherits from [`PreTrainedTokenizerFast`] which contains most of the main methods. Users should
    refer to this superclass for more information regarding those methods.
 
     Args:
```
Same here

**Review by sgugger — APPROVED** [review]
Very nice, just left a few nits for the docstrings.

For your issues with the quality scripts doing many changes, I think it might be due to a different version of black that we have pined? Make sure to run `pip install -e .[quality]` to have the right one. There are no files that should be modified on a fresh pull of the main branch since we enforce those checks at each commit.

**Comment by Davidy22** [comment]
Cleared up those little oversights.

Seems like the black version specified in setup.py is 22.3 and the one on my global python install is 22.6, so both should still be within the black project's promise of a yearly standard. The diff tells me that all the changes are just removing blank lines after function signatures or the first line of control flow blocks, which seems to be the usual black policy so I guess they just fixed some edge case that made it add a few empty lines mid-2022.

**Review by sgugger — APPROVED** [review]

**Comment by Davidy22** [comment]
I notice a little pile of deprecation warnings in that same leg of the test suite. If those aren't something that's being intentionally held back for compatibility reasons, I could put together another PR just mopping those up after all of the

**Comment by sgugger** [comment]
By all means! We are ignoring those for now, but we do need to clean them up at some point!

**Pull request merged**